fenixedu-ist-giaf-invoices/src/main/java/pt/ist/fenixedu/giaf/invoices/SapEvent.java

```diff
@@ -1081,13 +1081,32 @@ private DateTime getDocumentDate(DateTime documentDate, boolean isNewDate) {
 
     private boolean isToProcessDebt(boolean isGratuity) {
         return (isGratuity || event instanceof ExternalScholarshipPhdGratuityContribuitionEvent)
-                && !isGratuityBeforeOpenYear(event);
+                && !isGratuityCreatedAfterOpenYear(event);
     }
 
-    private static boolean isGratuityBeforeOpenYear(final Event event) {
+    private static boolean isGratuityCreatedAfterOpenYear(final Event event) {
         if (event.isGratuity()) {
-            final ExecutionYear executionYear = Utils.executionYearOf(event);
-            return executionYear.getEndCivilYear() < SapRoot.getInstance().getOpenYear();
+            final boolean hasDebt = event.getSapRequestSet().stream()
+                    .filter(sr -> sr.getRequestType() == SapRequestType.DEBT)
+                    .filter(sr -> sr.getIntegrated())
```

Comment:
Should we test this isIntegrated condition ?
When SAP requests a generated in sequence, first we create the DEBT then the CREDIT but both may be created in the same TX before tge DEBT is integrated.